File: frontend/src/app/dtos/seatmap.ts
import {Sector, SectorDto} from './sector';

export class Seatmap {
  id?: string;
  name?: string;
  sectors?: Sector[][];
  numberOfSectors?: number;
  numberOfSeats?: number;
  isUsed?: boolean;
}

export class SeatmapDto {
  id?: string;
  name: string;
  sectors: SectorDto[];
  numberOfSectors?: number;
  numberOfSeats?: number;
  isUsed?: boolean;
}

File: backend/src/main/java/at/ac/tuwien/sepm/groupphase/backend/entity/SeatMap.java
package at.ac.tuwien.sepm.groupphase.backend.entity;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Set;
import java.util.UUID;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class SeatMap {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private UUID id;

    private String name;

    @OneToMany(mappedBy = "seatMap", cascade = CascadeType.ALL, orphanRemoval = true)
    private Set<Sector> sectors;

    @OneToMany(mappedBy = "seatMap")
    private Set<Performance> performance;
}
